R/tax_range_strat.R

```diff
@@ -0,0 +1,225 @@
+#' Generate a stratigraphic section plot
+#'
+#' A function to plot the stratigraphic ranges of fossil taxa from occurrence
+#' data.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences containing
+#'   at least two columns: names of taxa, and their stratigraphic position (see
+#'   `name` and `level` arguments).
+#' @param name \code{character}. The name of the column you wish to be treated
+#'   as the input names, e.g. "genus" (default).
+#' @param level \code{character}. The name of the column you wish to be treated
+#'   as the stratigraphic levels associated with each occurrence, e.g. "bed"
+#'   (default) or "height". Stratigraphic levels must be \code{numeric}.
+#' @param certainty \code{character}. The name of the column you wish to be
+#'   treated as the information on whether an identification is certain (1) or
+#'   uncertain (0). By default (\code{certainty = NULL}), no column name is
+#'   provided, and all occurrences are assumed to be certain. In the plot,
+#'   certain occurrences will be plotted with a black circle and joined with
+#'   solid lines, while uncertain occurrences will be plotted with a white
+#'   circle and joined with dashed lines.
+#' @param by \code{character}. How should the output be sorted? Either: "FAD"
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()]. Useful arguments include `xlab` (the x-axis label),
+#'   `ylab` (the y-axis label, default is "Bed number"), `main` (the plot
+#'   title), `xlim` (the x-axis limits), and `ylim` (the y-axis limits). The
+#'   `axes` and `type` arguments are not supported and will be overridden.
```

Comment:
I think there are quite a lot of additional arguments that are not supported, for example, `pch`, `cex`, `col`, etc? If the purpose of this function is to just generate a quick plot, I think it is fine if we just clarify this. However, if the point is to have something highly customisable that can be used for publications, we probably want to re-look at this. Based on #99, this might be something that users request.

Reply:
I think this is something we can revisit at a later date, if/when someone raises it.

Reply:
Sure, I am happy with increasing customisability later. However, I think we still need to at least state which arguments can or cannot be used. Currently, the documentation suggests arguments available via `plot()` are compatible. If we don't do this, a lot of users will end up banging their head against the wall asking why very common arguments don't work (e.g. `col`). We might also get quite a few issues raised if this is not clear.

Reply:
I think the only arguments that are not currently supported are those related to the styling of the points and segments. It looks like these are all hard-coded at the moment. We could make these individual arguments for the function (although would we want different col and bg arguments for the certain and uncertain points?). Another option is to just have a list of arguments that would be passed to `points()` (again, would we want two arguments, one for the certain `points()` call, one for the uncertain `points()` call?). Yet another option is to have an argument that is a list of graphical parameters. In the function we could then save the old `par()`, set the new values with this argument, plot the points, then reload the old `par()`. And of course, a final option is to just report in the documentation that these things are fixed and can not be changed.

Reply:
I'm not sure that solves the problem of how to handle the certain vs uncertain point and line types? But if we're not terribly worried about that, then I'm fine pulling these arguments from `plot_args` if they are specified.

Reply:
What if by default the `col`, `bg`, `lty`, etc arguments are a vector of length 2, with the first element relating to certain occurrences and the second uncertain occurrences?